src/main/kotlin/graphics/scenery/volumes/Volume.kt

```diff
@@ -760,22 +766,38 @@ open class Volume(
 
             val start = System.nanoTime()
 
-//            if(reader.openPlane(0, 0).imageMetadata.isLittleEndian) {
             logger.debug("Volume is little endian")
-            (0 until reader.getPlaneCount(0)).forEach { plane ->
-                imageData.put(reader.openPlane(0, plane).bytes)
-            }
-//            } else {
-//                logger.info("Volume is big endian")
-//                (0 until reader.getPlaneCount(0)).forEach { plane ->
-//                    imageData.put(swapEndianUnsafe(reader.openPlane(0, plane).bytes))
-//                }
-//            }
+            val planeSize = bytesPerVoxel * dims.x * dims.y
+            if(hub.get<Settings>()?.get("Volume.ParallelReads", true) == true) {
+                val readers = ConcurrentHashMap<Thread, ReaderFilter>()
+                (0 until reader.getPlaneCount(0)).forEachIndexedAsync { index, plane ->
+                    val thread = Thread.currentThread()
+                    val myReader =
+                        readers.getOrPut(thread) { scifio.initializer().initializeReader(FileLocation(file.toFile())) }
```

Comment:
If we are putting explanation comments on companion objects incase someone does not know enough kotlin. These 5 lines need at least a paragrah. Because I know a bit of kotlin but am unable to understand it.

For each coroutine we take the current thread. That is which thread? A coroutine worker thread? And put or get it in some hashmap. Why get? And why use thread as the key?